WIP: Kaldi DNN demo

lipeng-unisound opened the pull request:
Works on a 1k utterance training set (about 260k+ samples), the frame error rate is below 5%. On a  99k utterance training set, the error remains to 70%-80%. I tried to increase the `pool_size` and `max_pool_size` parameters of provider function up to 1 million, but it doesn't help. Have no idea for now.

lipeng-unisound committed:
WIP: Kaldi DNN demo

Doesn't work yet. Error message:

[libprotobuf ERROR google/protobuf/io/coded_stream.cc:171] A protocol
message was rejected because it was too big (more than 67108864 bytes).

Signed-off-by: Li Peng <lipeng@unisound.com>

lipeng-unisound committed:
Fix a path error

Signed-off-by: Li Peng <lipeng@unisound.com>

lipeng-unisound committed:
Remove a redundant line

Signed-off-by: Li Peng <lipeng@unisound.com>

lipeng-unisound committed:
Fix the wrong module name

Signed-off-by: Li Peng <lipeng@unisound.com>

lipeng-unisound committed:
Fix some bugs

Now works for only small data, because the size of utt_pdf is restricted by
the google protobuf limit.

Run start-docker.sh, and then inside the docker container cd to
/paddle/demo/kaldi_dnn and run train-in-docker.sh.

Signed-off-by: Li Peng <lipeng@unisound.com>

lipeng-unisound commented:
Now works for only small data on CPU server.

lipeng-unisound committed:
Combine Kaldi feature with label into one ark file

In the previous commit, the labels (utt_pdf) are read in
trainer_config.dnn.py and passed via settings()'s args parameter. The
size of the labels is restricted by protobuf's limit, which is 64 MiB.
In this commit, I combine the feats.ark and pdf.txt into one
feats_pdf.ark file by the scripts combine-ark-label.py to solve this
problem.

Also in this commit, the two shell script for running with GPU is added.

Signed-off-by: Li Peng <lipeng@unisound.com>

pineking commented on demo/kaldi_dnn/start-docker.sh:
```diff
@@ -0,0 +1,7 @@
+set -e
+
+sudo docker run -it --rm \
+  -v /work/local/lipeng/github/unisound-ail/paddle/demo:/paddle/demo \
```
‘...lipeng...’ 里的路径写死了，用 `pwd` 之类

lipeng-unisound replied:
Done

pineking commented on demo/kaldi_dnn/gpu-start-docker.sh:
```diff
@@ -0,0 +1,9 @@
+
+sudo docker run -it --rm \
+  --privileged \
+  -v /home/core/lipeng/unisound-ail/paddle/demo:/paddle/demo \
```
‘...lipeng...’ 里的路径写死了，用 `pwd` 之类

lipeng-unisound replied:
Done

lipeng-unisound committed:
Obtain the demo directory automatically

Signed-off-by: Li Peng <lipeng@unisound.com>

wangkuiyi commented on demo/kaldi_dnn/gpu-start-docker.sh:
```diff
@@ -0,0 +1,10 @@
+  -v $work_dir:/paddle/demo \
+  -v /home/core/paddle/data:/paddle/data \
+  -v /var/lib/nvidia:/usr/local/nvidia/lib64 \
+  harbor.ail.unisound.com/liuqs_public/paddle:gpu-latest /bin/bash
```
harbor.ail.unisound.com 是外界访问不了的？这个pr的意图是不是一个demo？如果是用来展示技术肌肉的public demo的话，是不是docker image放到dockerhub.com上更合适？

lipeng-unisound replied:
我们没有修改 paddle。这个 PR 是想先用一个我们熟悉的网络结构（DNN）和数据（Kaldi 格式的语音特征和 label）来尝试把 Paddle 用对。当前的 repo 是 private 的，仅内部交流。

lazzylee committed:
Fix bad performance when cache=CacheType.NO_CACHE in dataprovider

cache=CacheType.CACHE_PASS_IN_MEM: train and eval performance is similar
cache=CacheType.NO_CACHE: eval's performance is much worse
Issue 2: https://github.com/unisound-ail/Paddle/issues/4#issuecomment-258757154

lazzylee committed:
Add data download pre-process
